### PR TITLE
Add version switcher, favicon, and context-dependent logo

### LINKED ISSUE #584: add version switcher to online docs
We used to be able to switch between the "stable" and "latest" version of the online docs. 
Now we can only do it by changing the url manually. 

It would be nice to add the switcher back, following the [explanations](https://pydata-sphinx-theme.readthedocs.io/en/stable/user_guide/version-dropdown.html) from PyData theme.

### PR DESCRIPTION
**Pull request by nwlandry**
Fixes #584 and #585.

This PR
* Adds a favicon
* Creates a dark and light version of the logo
* Adds a version switcher

### CODE REVIEW

**Comment by maximelucas** [comment]
Amazing Nich!! Favicon is a sweet plus.

Two things: 
1. On my laptop the menu is now looking a bit strange because on two lines
2. the light theme logo doesn't have a margin anymore so it's stuck to the "new release" button (dark theme logo doesn't have that problem)

image

I'm not fully sure how to solve 1., I think we need to find a way to control the width of the[ 3 navbars/zones (left, middle, right)](https://pydata-sphinx-theme.readthedocs.io/en/stable/user_guide/layout.html)? Can the version switcher be made less wide by changing the default text from "choose version" to "vesion" or simply "stable" by default?

**Comment by nwlandry** [comment]
In response to the review I
* reverted the logo back
* adjusted the spacing in the navbar
* Added a call to the Github API to get the date of the latest release

I believe I addressed all your comments, @maximelucas!

File: docs/source/conf.py [hunk 1]
```diff
@@ -115,9 +118,18 @@
 
 # There are two options for replacing |today|: either, you set today to some
 # non-false value, then it is used:
-# today = ''
+r = requests.get("https://api.github.com/repos/xgi-org/xgi/releases/latest")
+
+if r.ok:
+    release_date = (
+        datetime.fromisoformat(r.json()["published_at"]).date().strftime("%B %d, %Y")
```
Comment (maximelucas) [inline docs/source/conf.py:125]:
```suggestion
        datetime.fromisoformat(r.json()["published_at"]).date().strftime("%b %d, %Y")
```

File: docs/source/conf.py [hunk 2]
```diff
@@ -184,6 +201,13 @@
             "icon": "fa-brands fa-mastodon",  # Font Awesome icon
         },
     ],
+    "header_links_before_dropdown": 4,
```
Comment (maximelucas) [inline docs/source/conf.py:204]: Anyway of having 5 and keeping the Gallery visible (not in dropdown)?

File: docs/source/conf.py (hunk 1 above)
Comment (maximelucas) [reply docs/source/conf.py:125]: just short month version for space

**Comment by maximelucas** [comment]
You're on fire, the automatic date retrieval is great!!
Two tiny comments but I'll let you choose, approving already.

**Review by maximelucas — APPROVED** [review]

File: docs/source/conf.py (hunk 2 above)
Comment (nwlandry) [reply docs/source/conf.py:204]: The issue was that I wanted everything on one line even when the user wasn't in full-screen, which is basically impossible unless the navbar is slimmed down. So I will keep it like it is for now, and then change it later if need be.

**Comment by nwlandry** [comment]
Thanks for the review!

**Pull request merged**